docs/source/filters.rst

```diff
@@ -3,68 +3,125 @@
+:ref:`Filters and Actions reference <policy>` for
+aws resource-specific filters.
 
-Value Filters
+Value Filter
 -------------
 
 Cloud Custodian provides for a flexible query language on any resource by
 allowing for rich queries on JSON objects via JMESPath, and allows for
-mixing and combining those with a boolean conditional operators that
-are nest-able. Comparison between values is configurable supporting
-scalar operators:
+mixing and combining those with boolean conditional operators that
+are nest-able. (Tutorial here on `JMESPath <http://jmespath.org/tutorial.html>`_ syntax)
+
+
+The base value filter enables the use of jmespath with data returned from a describe call.
+
+.. code-block:: yaml
+
+    filters:
+         - type: value
+           key: "State[0]"    ─▶ The value from the describe call
+           value: "running"   ─▶ Value that is being filtered against
+
+
+There are several ways to get a list of possible keys for each resource.
+
+- Via Custodian CLI
+
+    Create a new custodian yaml file with just the name and resource fields. Then run
+    ``custodian run -s OUTPUT_DIR``. The valid key fields can be found in the output directory
+    in resources.json
+
+    .. code-block:: yaml
+
+        policies:
+          - name: my-first-policy
+            resource: aws.ec2
+
+- Via Cloud Providers CLI
+
+    Use the relevant cloud provider cli to run the describe call to view all available keys. For example
+    using aws cli run ``aws ec2 describe-instances`` or with azure ``az vm list``.
+
+    Note: You do not need to include the outermost json field in most cases since custodian removes this field
+    from the results.
+
+- Via Cloud Provider Documentation
+
+    Go to the relevant cloud provider sdk documentation and search for the describe api call for the resource
+    your interested in. The available fields will be listed under the results of that api call.
+
+
 
 - Comparison operators:
+    The generic value filter allows for comparison operators to be used
+
     - ``equal`` or ``eq``
     - ``not-equal`` or ``ne``
     - ``greater-than`` or ``gt``
     - ``gte`` or ``ge``
     - ``less-than`` or ``lt``
     - ``lte`` or ``le``
-- Other operators
+
+  .. code-block:: yaml
+
+      filters:
+         - type: value
+           key: CpuOptions.CoreCount      ─▶ The value from the describe call
+           value: 36                      ─▶ Value that is being compared
+           op: greater-than               ─▶ Comparison Operator
+
+- Other operators:
     - ``absent``
     - ``present``
     - ``not-null``
     - ``empty``
-- Collection operators against user supplied lists:
+
+  .. code-block:: yaml
+
+      filters:
+         - type: value
+           key: CpuOptions.CoreCount      ─▶ The value from the describe call
+           value: present                 ─▶ Checks if key is present
+
+
+- List Operators:
+    There is a collection of operators that can be used with user supplied lists:
```

Comment:
There's some subtle distinctions on 'in' / 'not-in' and the need for value_type: swap that should be here.

note `contains` also avoids the need for value_type: swap

ie. which side is the list and which side is the scalar.

Reply:
also your including a bunch of boolean operators under list-operators